### PR TITLE
[REVIEW] Fix docs for `value_counts` and update test cases

### LINKED ISSUE #4431: [DOC] value_counts description is incorrect
## Report incorrect documentation

**Location of incorrect documentation**
https://rapidsai.github.io/projects/cudf/en/0.12.0/api.html?highlight=value_counts#cudf.core.series.Series.value_counts

**Describe the problems or issues found in the documentation**
Description of the function was copied from uniques().  Should be closer to:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.value_counts.html?highlight=value_counts#pandas.Series.value_counts

**Steps taken to verify documentation is incorrect**
Checked Pandas documentation

**Suggested fix for documentation**
Copy Pandas description.

---

## Report needed documentation

**Report needed documentation**
Return a Series containing counts of unique values.

The resulting object will be in descending order so that the first element is the most frequently-occurring element. Excludes NA values by default.

**Describe the documentation you'd like**
What's the difference between "needed documentation and documentation you'd like?"

**Steps taken to search for needed documentation**
List any steps you have taken:
google search

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
__setitem__ was missing couple of scalar handling which was present in earlier version of cudf.

also updates doc for value_counts for cudf.Series.

closes #4431

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -600,14 +600,16 @@ def __setitem__(self, arg, value):
...
-                        value = column.as_column(value)
                     for col in arg:
                         # we will raise a key error if col not in dataframe
                         # this behavior will make it
                         # consistent to pandas >0.21.0
                         if not is_scalar(value):
-                            self._data[col] = value
+                            self._data[col] = column.as_column(value)
+                        elif col not in self._data.keys():
+                            self._data[col] = column.as_column(
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py]: Let's use `column.as_column(value, length=len(self))` instead of calling `utils.scalar_broadcast_to(...)`.

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/dataframe.py]: `as_column` is also calling broadcast, but that would make future proof

File: python/cudf/cudf/tests/test_setitem.py [hunk 2]
```diff
@@ -45,6 +45,23 @@ def test_dataframe_setitem_existing_columns(df, arg, value):
...
+        (DataFrame(), ["a", "b"], 0, DataFrame({"a": [], "b": []})),
+        (
+            DataFrame({"a": [1, 2, 3]}),
+            ["b"],
+            9,
+            DataFrame({"a": [1, 2, 3], "b": [9, 9, 9]}),
+        ),
+    ],
+)
+def test_dataframe_setitem_non_existing_columns(df, arg, value, expected):
```
Comment (shwina) [inline python/cudf/cudf/tests/test_setitem.py]: Shouldn't we be testing against Pandas here?

File: python/cudf/cudf/tests/test_setitem.py [hunk 3]
```diff
@@ -45,6 +45,23 @@ def test_dataframe_setitem_existing_columns(df, arg, value):
     assert_eq(df, gdf, check_dtype=False)
 
 
+@pytest.mark.parametrize(
+    ("df", "arg", "value", "expected"),
+    [
+        (DataFrame(), ["a", "b"], 0, DataFrame({"a": [], "b": []})),
```
Comment (shwina) [inline python/cudf/cudf/tests/test_setitem.py]: To be clear, are we testing the following? If so, it feels wrong...

```python
df = DataFrame()
df[["a", "b"]] = 0
assert_eq(df, DataFrame({"a": [], "b": []})
```

File: python/cudf/cudf/tests/test_setitem.py (hunk 2 above)
Comment (rgsl888prabhu) [reply python/cudf/cudf/tests/test_setitem.py]: pandas throws error in this case, but this was supported by old cudf before I made changes `__settitem__`.

File: python/cudf/cudf/tests/test_setitem.py (hunk 3 above)
Comment (rgsl888prabhu) [reply python/cudf/cudf/tests/test_setitem.py]: This was supported by earlier version of cudf, but pandas throws error with similar approach.

Comment (shwina) [reply python/cudf/cudf/tests/test_setitem.py]: It seems to me like it's a bug that we should no longer "support" :) Is there anywhere within cudf that we depend on this behaviour?

Comment (rgsl888prabhu) [reply python/cudf/cudf/tests/test_setitem.py]: No we don't, it was request from user in issue #4657

**Pull request merged**